robot/packages/cloud_package_manager.go

```diff
@@ -438,10 +450,8 @@ func (m *cloudManager) unpackFile(ctx context.Context, fromFile, toDir string) e
 
 	tarReader := tar.NewReader(archive)
 	for {
-		select {
-		case <-ctx.Done():
-			return errors.New("interrupted")
-		default:
+		if err := ctx.Err(); err != nil {
+			return err
```

Comment:
Happy to revert if people are opposed.

Reply:
This no longer returns "interrupted", unsure if that was important, but I like this check much better.

Reply:
Agreed. We had some old code that did the select{} style, and it makes sense when you have other channels, but then it got copy-pasted to cases NOT using other channels, and then plenty of other people just copied that. Definitely ctx.Err() is cleaner, IMHO.